lib/BasicAnalyses/CSVAliasAnalysis.cpp

```diff
@@ -0,0 +1,109 @@
+/// \file CSVAliasAnalysis.cpp
+/// \brief Decorate memory accesses with information about CSV aliasing.
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LLVMContext.h"
+#include "llvm/IR/MDBuilder.h"
+#include "llvm/IR/PassManager.h"
+#include "llvm/IR/Value.h"
+
+#include "revng/BasicAnalyses/CSVAliasAnalysis.h"
+#include "revng/Support/Assert.h"
+
+using namespace llvm;
+using CSVAAP = CSVAliasAnalysisPass;
+
+PreservedAnalyses CSVAAP::run(Function &F, FunctionAnalysisManager &FAM) {
+  // Get the result of the GCBI analysis
+  auto &M = *F.getParent();
+  const auto &MAMProxy = FAM.getResult<ModuleAnalysisManagerFunctionProxy>(F);
+  GCBI = MAMProxy.getCachedResult<GeneratedCodeBasicInfoAnalysis>(M);
+  if (!GCBI)
+    GCBI = &(FAM.getResult<GeneratedCodeBasicInfoAnalysis>(F));
```

Comment:
Is this pass actually being run both from a FunctionPassManager and a (Module)PassManager?

Reply:
Correct.

Reply:
OK, check the other ones too. I don't want this pattern to become ubiquitous.